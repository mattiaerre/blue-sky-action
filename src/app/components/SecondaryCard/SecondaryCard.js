import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const SecondaryCard = ({ article }) => (
  <div className="card">
    {article.publishedAt && (
      <div className="card-header">{moment(article.publishedAt).fromNow()}</div>
    )}
    <div className="card-body">
      <h4 className="card-title">
        <a
          href={article.url}
          target="_blank"
          rel="noopener noreferrer"
          dangerouslySetInnerHTML={{ __html: article.title }}
        />
      </h4>
      <p className="card-text">{article.description}</p>
    </div>
  </div>
);

SecondaryCard.propTypes = {
  article: PropTypes.object.isRequired
};

export default SecondaryCard;
