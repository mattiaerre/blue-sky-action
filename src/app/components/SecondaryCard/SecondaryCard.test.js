import React from 'react';
import { create } from 'react-test-renderer';
import SecondaryCard from './SecondaryCard';

describe('<SecondaryCard />', () => {
  const props = {
    article: {
      description:
        'Your daily roundup of the biggest TechCrunch stories and startup news.',
      publishedAt: '2017-10-31T20:00:33Z',
      title: 'Crunch Report',
      url: 'https://techcrunch.com/video/crunchreport/'
    }
  };
  test.only('to match snapshot', () => {
    const tree = create(<SecondaryCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
