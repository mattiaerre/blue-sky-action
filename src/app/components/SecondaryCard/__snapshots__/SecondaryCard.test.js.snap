// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryCard /> to match snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    2 months ago
  </div>
  <div
    className="card-body"
  >
    <h4
      className="card-title"
    >
      <a
        dangerouslySetInnerHTML={
          Object {
            "__html": "Crunch Report",
          }
        }
        href="https://techcrunch.com/video/crunchreport/"
        rel="noopener noreferrer"
        target="_blank"
      />
    </h4>
    <p
      className="card-text"
    >
      Your daily roundup of the biggest TechCrunch stories and startup news.
    </p>
  </div>
</div>
`;
